src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -286,6 +286,16 @@ protected virtual void Dispose(bool finalize)
             if (!finalize)
             {
                 this.ClearSolutionData();
+
+                // tell event listener to stop listening since this workspace is going away
```

Comment:
I am liking this if for no other reason in that it means we can decouple things better -- the features pulling off of the workspace are now decoupled from having to be initialized in the core workspace.